tokio/src/task/local.rs

```diff
@@ -623,12 +625,14 @@ impl Drop for LocalSet {
                 task.shutdown();
             }
 
+            // We already called shutdown on all tasks above, so there is no
+            // need to call shutdown.
             for task in self.context.tasks.borrow_mut().queue.drain(..) {
-                task.shutdown();
+                drop(task);
```

Comment:
This was changed a bunch of places. I don't see a drop impl added on the type? Why is this changed?

Reply:
The type in question is `Notified`, and it already has a destructor. I changed it because otherwise you need to go through `assert_owner` to call `shutdown`. The call to `shutdown` has been unnecessary since #3955, as the inject queue and similar can no longer hold newly spawned tasks.